app/controllers/api/v1/contacts_controller.rb

```diff
@@ -13,6 +13,9 @@ def create
         role = Role.find_or_create_by(title: role)
         ContactRole.create(contact_id: contact.id, role_id: role.id)
       end
+      role = Role.find_or_create_by(title: contact_params[:job_title])
+      ContactRole.create(contact_id: contact.id, role_id: role.id)
```

Comment:
I think I understand what is going on but I am not 100% positive. So after we create the contact, we go through each role where we find that role based on the title, and then we create a ContactRole once the role is defined for each given role. Then after that we are redefining the role based the contact_params passed in and create a whole new ContactRole with that information correct?

Reply:
correct! that is exactly how it is functioning